src/inputController.ts

```diff
@@ -31,35 +31,40 @@ export class InputController {
                 const data: IFileCreatorInputData = {
                     inputName,
                     pathToCreateAt: this.directoryPathToCreateAt,
-                    templateFolderPath: this.templateFolderPath,
-                    templateName,
+                    templateFolderPath: templateDirectory,
                 };
                 return data;
             });
 
     }
 
-    private showTemplatePickerDialog(templateFolderPath: string): Promise<string> {
+    private showTemplatePickerDialog(templateFolderPath: string[]): Promise<string> {
         return new Promise((resolve, reject) => {
-
-            let templateNames: string[];
-            try {
-                templateNames = this.availableTemplateNames(templateFolderPath);
-            } catch (error) {
-                // tslint:disable-next-line:max-line-length
-                reject(new Error(`${constants.ERROR_SETUP_MESSAGE_PREFIX} Could not find folder: ${templateFolderPath}. Please see ${constants.README_URL} for information on setting up Blueprint in your project.`));
-                return;
+            let templates: string[] = [];
+
+            for (const templatePath of templateFolderPath){
+                let templateNames: string[];
+                try {
+                    templateNames = this.availableTemplateNames(templatePath);
+
+                    const templateObject: string[] = templateNames.map((str) => templatePath + "\\" + str);
```

Comment:
Let's use path.join here to make sure we are OS agnostic (we already had to make that change for Windows support a little bit ago).